src/main/scala/com/wix/async/AsyncExecution.scala

```diff
@@ -74,7 +86,8 @@ protected[async] class AsyncExecution[T](executorService: ExecutorService,
     val elapsedInBlockingCall = Stopwatch.start()
     val res: T = nested
     val duration = elapsedInBlockingCall()
-    if (timeout != Duration.Zero && duration > timeout) {
+
+    if ((timeout != Duration.Zero && duration > timeout) || timedOut) {
```

Comment:
Basically, I'd suggest to transform this line to `timeout != Duration.Zero && timedOut`. The problem with initial implementation can be described by the quote from https://blogs.oracle.com/dholmes/entry/inside_the_hotspot_vm_clocks:
> In Java, time-triggered events are performed (at the lowest-level) through methods like Object.wait(millis, nanos), Thread.sleep(millis, nanos) and the java.util.concurrent.locks.LockSupport.park methods. These latter methods underpin the concurrency utilities. Although there are issues with the legacy methods (wait, sleep) rounding (up or down) the millisecond value based on the nanosecond value, ultimately the times are passed through to the platform specific mechanism for doing a timed "wait" - and the same mechanism is typically used by all methods (primarily for consistent support of thread interruption). On Solaris and Linux these calls take time structures that allow microsecond, or nanosecond, values to be passed - but there is no guarantee of that resolution being achieved: In fact we know it won't be because of the typical timer interrupt rate of 10ms - so the operating system applies its own rounding algorithms internally to those calls

Therefore - previously we had two different answers on whether we are timed out ("we've got TimeoutException and reported GaveUp" or "we exceeded duration and reported ExceededTimeout"). I would suggest to turn those into one answer - "we've got TimeoutException".